src/test/java/org/springframework/data/couchbase/repository/query/PartTreeN1qBasedQueryTest.java

```diff
@@ -112,17 +114,50 @@ public void testGetCountExcludesDynamicSortClause() throws Exception {
 		PartTreeN1qlBasedQuery query = new PartTreeN1qlBasedQuery(queryMethod, couchbaseOperations);
 		Statement statement = query.getCount(accessor, new Object[] { "value", pr });
 
-		assertEquals("SELECT COUNT(*) AS count FROM `default` WHERE name = \"value\" "
+		assertEquals("SELECT COUNT(*) AS count FROM `default` WHERE (name = \"value\") "
 				+ "AND `_class` = \"org.springframework.data.couchbase.core.Beer\"", statement.toString());
 
 	}
 
+
+	@Test
+	public void testWrapWhereCriteria() throws Exception {
+		CouchbaseOperations couchbaseOperations = mock(CouchbaseOperations.class);
+		CouchbaseBucket couchbaseBucket = mock(CouchbaseBucket.class);
+		CouchbaseConverter couchbaseConverter = mock(CouchbaseConverter.class);
+		ParameterAccessor accessor = mock(ParameterAccessor.class);
+		ProjectionFactory factory = mock(ProjectionFactory.class);
+
+
+		MappingContext mappingContext = new CouchbaseMappingContext();
+		Method method = TestRepository.class.getMethod("findByIdOrName", String.class, String.class);
+		RepositoryMetadata metadata = new DefaultRepositoryMetadata(TestRepository.class);
+		CouchbaseQueryMethod queryMethod = new CouchbaseQueryMethod(method, metadata, factory, mappingContext);
+
+		when(couchbaseOperations.getCouchbaseBucket()).thenReturn(couchbaseBucket);
+		when(couchbaseBucket.name()).thenReturn("B");
+		when(couchbaseOperations.getConverter()).thenReturn(couchbaseConverter);
+		when(couchbaseOperations.getConverter().getMappingContext()).thenReturn(mappingContext);
+		when(couchbaseConverter.getTypeKey()).thenReturn("_class");
+		when(accessor.iterator()).thenReturn(Arrays.asList((Object) "value1", (Object) "value2").iterator());
+		when(couchbaseConverter.convertForWriteIfNeeded(eq("value1"))).thenReturn("value1");
+		when(couchbaseConverter.convertForWriteIfNeeded(eq("value2"))).thenReturn("value2");
+
+
+		PartTreeN1qlBasedQuery query = new PartTreeN1qlBasedQuery(queryMethod, couchbaseOperations);
+		Statement statement = query.getStatement(accessor, null);
+
+		assertEquals("SELECT META(`B`).id AS _ID, META(`B`).cas AS _CAS, `B`.* FROM `B` WHERE (`id` = \"value1\" OR `name` = \"value2\")"
+				+ " AND `_class` = \"org.springframework.data.couchbase.core.Beer\"", statement.toString());
+	}
+
 	public static interface TestRepository extends CrudRepository<Beer, String> {
 
 		Page<Beer> findByNameOrderByName(String name, Pageable pageRequest);
 
 		Page<Beer> findByName(String name, Pageable pageRequest);
 
+		List<Beer> findByIdOrName(String id, String name);
```

Comment:
I would probably also test in an integration test that that sort of query does indeed return the expected results from db when executed from end to end...